atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -254,43 +275,38 @@ int getMaxTriesTotal() {
     private synchronized void refreshPool() {
         blacklist.checkAndUpdate(cassandra.getPools());
 
-        Set<InetSocketAddress> serversToAdd = Sets.newHashSet(config.servers());
-        Set<InetSocketAddress> serversToRemove = ImmutableSet.of();
-
         if (config.autoRefreshNodes()) {
-            serversToAdd.addAll(cassandra.refreshTokenRanges());
+            setServersInPoolTo(cassandra.refreshTokenRangesAndGetServers());
+        } else {
+            setServersInPoolTo(config.servers());
```

Comment:
Is there any case where auto refresh should be off? Not in scope for this PR, but should we support it?

Reply:
I can't imagine why we'd want that, but maybe cassops people can think of a scenario